services/registration/src/routes/application.ts

```diff
@@ -78,7 +78,7 @@ applicationRouter.route("/actions/choose-application-branch").post(
     });
 
     if (existingApplication) {
-      if (existingApplication.applied) {
+      if (existingApplication.status === StatusType.APPLIED) {
```

Comment:
In this scenario, you will also need to check the other states. For example, if a user is confirmed, rejected, or applied, they shouldn't be able to choose an application branch.